src/macro/macroApi.ml

```diff
@@ -1747,6 +1751,56 @@ let macro_api ccom get_api =
 			(get_api()).add_global_metadata (decode_string s1) (decode_string s2) (decode_bool b1,decode_bool b2,decode_bool b3) (get_api_call_pos());
 			vnull
 		);
+		"register_define_impl", vfun2 (fun d src ->
+			let flags : define_parameter list = [] in
+
+			let platforms = decode_opt_array decode_string (field d "platforms") in
+			let flags =
+				if (List.length platforms) = 0 then flags
```

Comment:
Note that it's generally better to match the list against `[]` for this because `length` is O(n).